staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -149,6 +149,20 @@ const (
 	//
 	// Allows sending warning headers in API responses.
 	WarningHeaders featuregate.Feature = "WarningHeaders"
+
+	// owner @smarterclayton
+	// alpha: v1.16
+	// beta:  v1.19
+	//
+	// Enable legacy behavior to vary cluster functionality on the node-role.kubernetes.io labels. On by default (legacy), will be turned off in 1.18.
+	LegacyNodeRoleBehavior featuregate.Feature = "LegacyNodeRoleBehavior"
```

Comment:
Ideally there's a better hope for these but I think this will be okay for v1.19

Reply:
Wouldn't be opposed to a features package like this one in k8s.io/cloud-provider, not sure if there's been discussions around a centralized place for all feature gates.

Reply:
The only features that should go in this file are ones that apply to all API servers built on this framework, not anything specific to kube-apiserver / kube-controller-manager, etc.

Scanning the features above, WinOverlay/WinDSR do not belong in this file. The other features are generic API server gates.

Reply:
cleaned those two feature gates up in https://github.com/kubernetes/kubernetes/pull/92871

Reply:
For the feature flags to be consumable by cloud provider code it should live in a public space and not the private pkg/features. If we believe that k8s.io/apiserver is the wrong public repo, then we should decide what we believe to be the correct public repo. k8s.io/component-base (I think its low enough in the dependency graph)? apimachinery?

Reply:
How about moving the definition of the features (not the feature gate instance) to k8s.io/api?  These gates are logically bound to the kube version.  And by keeping the featuregate instance in component-base, we can have a way to "register all the featuregates for your process" without being opinionated about what those featuregates are.  This allows a consistent source of kube featuregate definitions and allows reuse of component-base by things outside of kube.

Reply:
+1 for anything as high as k8s.io/api in the graph. Component-base and apimachinery are far too low. They are supposed to be generic. We shouldn't work against that direction.

Reply:
Feature gates that aren't defined cause panics, https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L302 and if a constant is removed, the code won't compile.  I think this protects against "I depended on a kube feature that was removed" and would still allow a binary to add back ones it wanted.  It also gives a positive failure on lookup values so developers learn quickly.

Reply:
Updated PR to move CCM related feature gates to cmd/controller-manager(which will be public soon https://github.com/kubernetes/kubernetes/pull/91014) due to Type Feature is declared in k8s.io/component-base. Please take a look and let me know if we would still prefer to keep them in k8s.io/api (which means we would also move Type Feature Declaration around). Thank you

Reply:
I think this set of changes should be removed in the current iteration.